### DISCUSSION TITLE
MX25L12835F_FW_20210306_retail.BIN

### DISCUSSION THREAD

**Discussion by bronken** [ludwig-v/wireless-carplay-dongle-reverse-engineering#59] [Show and tell]
I have made a working dump from the newest flash. Restored a friends dead cplay2air with it. Anybody needs it? 
I cannot add it to the repo.

**Comment by CiNcH83** [comment]
Amazing how many people brick their devices. The updating process does not seem to be too stable. You could only restore it with a programmer right?

**Reply by bronken** [reply]
I dont know if there were other ways to do this but cloning the Flash from another Carplay Dongle solved it for me.

**Comment by ludwig-v** [comment]
Make a pull request and add your dump inside the already existing folder with my dumps ;)
Or make it downloadable and I will add it

**Reply by bronken** [reply]
https://www.dropbox.com/s/gbogit5qy9jskjz/MX25L12835F_FW_20210306_retail.BIN?dl=0

**Reply by jsm174** [reply]
just curious, was the dump done clipped to the device? When I tried to do that, I think I killed mine.  https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/issues/8#issuecomment-742146406

**Reply by bronken** [reply]
Desoldered. But clipping works as well but you always have to remember to short circuit the xtal of the cpu. Giving the flash or ee power will, in most cases, give power to the cpu in reverse. Then the cpu boots and both your programmer and the cpu will try to talk to the flash or ee which then leads to unusable data.

**Reply by zkSac** [reply]
> https://www.dropbox.com/s/gbogit5qy9jskjz/MX25L12835F_FW_20210306_retail.BIN?dl=0

Can u re upload pls! Thank you so much!

**Reply by leonardosalles** [reply]
> > https://www.dropbox.com/s/gbogit5qy9jskjz/MX25L12835F_FW_20210306_retail.BIN?dl=0
> 
> Can u re upload pls! Thank you so much!

It is inside the "Flash_Dump" folder https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/tree/master/Flash_Dump

**Reply by qdk0901** [reply]
> just curious, was the dump done clipped to the device? When I tried to do that, I think I killed mine. [#8 (comment)](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/issues/8#issuecomment-742146406)

I think you could have missed the OTP part on the flash

**Reply by jsm174** [reply]
> 
> I think you could have missed the OTP part on the flash 
> 
> 

Missed?

**Reply by qdk0901** [reply]
@jsm174 I have successfully switched the carlinkit dongle from U2W mode to AutoBox mode(which you mentioned as 'android based' mode)，just flash this file
[modified.zip](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/files/7577842/modified.zip)
), the dropbear stuffs are built into the image.

I will make a tutorial about how to construct the flash image when I have time.

**Reply by jsm174** [reply]
Awesome! Eager to see your tutorial!

**Reply by qba667** [reply]
@qdk0901 can you provide some details on this? I got the V4 adapter, wondering if I can downgrade it using this method.

**Reply by Ondrashek** [reply]
Hi @bronken 
Do you know where I can do short circuit the xtal pin on main cpu on V4 hardware? I am not able to flash chip again with clips anymore (cannot find chip or shows me unknown chip)

Thanks :-)

**Comment by bronken** [comment]
I desordered the flash. I always do bc on some pcbs you wake up the cpu with your vcc and then you habe two spi masters talking.
So desolder (preferred) or at least short circuit the crystal of the main cpu.
